astroquery/desi/core.py

```diff
@@ -0,0 +1,111 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+import urllib.error
+import requests
+import warnings
+
+import pyvo as vo
+import numpy as np
+import astropy.coordinates as coord
+
+from astropy import units as u
+from astroquery.exceptions import NoResultsWarning
+from astroquery.query import BaseQuery
+from astroquery.utils import commons
+from astroquery.desi import conf
+
+__all__ = ['DESILegacySurvey', 'DESILegacySurveyClass']
+
+
+class DESILegacySurveyClass(BaseQuery):
+
+    def query_region(self, coordinates, radius=0.5*u.arcmin, *, data_release=9):
```

Comment:
There was a suggestion earlier to make `radius` an optional keyword argument, like this:
```suggestion
    def query_region(self, coordinates, *, radius=0.5*u.arcmin, data_release=9):
```
I resolved the conversation unintentionally, but I'll re-raise it now: is there any reason not to have it be optional?

Reply:
In some modules the region can be a circle or a rectangle, and in those cases `radius` should definitely be a keyword-only argument. Although here the region can only be a circle, it might still better to make it a keyword-only argument for consistency's sake with the other sub-packages, and to make it very explicit to all users that the region is indeed a circle and not a rectangle.

Reply:
...actually it is a (projected) square, so `radius` is the wrong kwarg.
```suggestion
    def query_region(self, coordinates, width=0.5*u.arcmin, *, data_release=9):
```
https://github.com/astropy/astroquery/pull/2210/files#diff-79317ce77daf56401a2a4150461a92c123e7061152b9219d313ac47515f9ad2aR44-R48

Probably this should be refactored to have `width` and `height`, and maybe even specifically `ra_width` and `dec_height`, as arguments.